### PR TITLE
Add tests and update Api wrapper

### LINKED ISSUE #13: Overhaul testing
Check if tests are "best practice" and if they can be enhanced in some way.

These functions are important for us:
#### Managing backup tasks
- [x] `backup_task_get`
      https://github.com/SAPUCC/sepsesam/blob/6f1c3eb35f155f30b6c935d4bb60cf3706e5328a/src/sepsesam/api.py#L842
- [x] `backup_task_create`
      https://github.com/SAPUCC/sepsesam/blob/6f1c3eb35f155f30b6c935d4bb60cf3706e5328a/src/sepsesam/api.py#L873
- [x] `backup_task_delete`
      https://github.com/SAPUCC/sepsesam/blob/6f1c3eb35f155f30b6c935d4bb60cf3706e5328a/src/sepsesam/api.py#L909

#### Managing clients
- [x] `client_find`
      https://github.com/SAPUCC/sepsesam/blob/6f1c3eb35f155f30b6c935d4bb60cf3706e5328a/src/sepsesam/api.py#L242
- [x] `client_create`
      https://github.com/SAPUCC/sepsesam/blob/6f1c3eb35f155f30b6c935d4bb60cf3706e5328a/src/sepsesam/api.py#L283
- [x] `client_update`
      https://github.com/SAPUCC/sepsesam/blob/6f1c3eb35f155f30b6c935d4bb60cf3706e5328a/src/sepsesam/api.py#L301
- [x] `client_delete`
      https://github.com/SAPUCC/sepsesam/blob/6f1c3eb35f155f30b6c935d4bb60cf3706e5328a/src/sepsesam/api.py#L324

#### Manage external groups
- [x] `external_group_find`
      https://github.com/SAPUCC/sepsesam/blob/6f1c3eb35f155f30b6c935d4bb60cf3706e5328a/src/sepsesam/api.py#L1510
- [x] `external_group_create`
      https://github.com/SAPUCC/sepsesam/blob/6f1c3eb35f155f30b6c935d4bb60cf3706e5328a/src/sepsesam/api.py#L1516
- [x] `external_group_delete`
      https://github.com/SAPUCC/sepsesam/blob/6f1c3eb35f155f30b6c935d4bb60cf3706e5328a/src/sepsesam/api.py#L1539

#### Manage groups
- [x] `group_create`
      https://github.com/SAPUCC/sepsesam/blob/6f1c3eb35f155f30b6c935d4bb60cf3706e5328a/src/sepsesam/api.py#L1347
- [x] `group_delete`
      https://github.com/SAPUCC/sepsesam/blob/6f1c3eb35f155f30b6c935d4bb60cf3706e5328a/src/sepsesam/api.py#L1372

#### Manage locations
- [x] `location_resolve_to_id`
      https://github.com/SAPUCC/sepsesam/blob/6f1c3eb35f155f30b6c935d4bb60cf3706e5328a/src/sepsesam/api.py#L444
- [x] `location_create`
      https://github.com/SAPUCC/sepsesam/blob/6f1c3eb35f155f30b6c935d4bb60cf3706e5328a/src/sepsesam/api.py#L388
- [x] `location_delete`
      https://github.com/SAPUCC/sepsesam/blob/6f1c3eb35f155f30b6c935d4bb60cf3706e5328a/src/sepsesam/api.py#L429

#### Manage commands
- [x] https://github.com/SAPUCC/sepsesam/blob/6f1c3eb35f155f30b6c935d4bb60cf3706e5328a/src/sepsesam/api.py#L1701

#### Manage command events
- [x] https://github.com/SAPUCC/sepsesam/blob/6f1c3eb35f155f30b6c935d4bb60cf3706e5328a/src/sepsesam/api.py#L1773

#### Manage backup schedules
- [x] https://github.com/SAPUCC/sepsesam/blob/6f1c3eb35f155f30b6c935d4bb60cf3706e5328a/src/sepsesam/api.py#L1630

#### Manage drive groups
- [x] https://github.com/SAPUCC/sepsesam/blob/6f1c3eb35f155f30b6c935d4bb60cf3706e5328a/src/sepsesam/api.py#L1845

### PR DESCRIPTION
**Pull request by DerYannis1**
add more tests and updated some functions of the api wrapper
closes #13 
Enhancements regarding #5

### CODE REVIEW

File: t/tests.py [hunk 1]
```diff
@@ -10,50 +10,116 @@
...
     """ sepsesam.api unittests """
 
-    """ server info """
 
-    def testGetServerInfo(self):
-        self.assertEqual(api.get_server_info()["restPort"], 11401)
+    """Managing Backup Tasks"""
+
+    
+
```
Comment (v411e) [inline t/tests.py]: Please consider using a formatter like [Ruff](https://marketplace.visualstudio.com/items?itemName=charliermarsh.ruff) for auto-formatting. These line breaks are huuge :)

**Review by v411e — CHANGES_REQUESTED** [review]
Looks good, thanks a lot! I'd be glad if you could auto-format it before we merge it

File: src/sepsesam/api.py [hunk 2]
```diff
@@ -498,7 +498,7 @@ def acl_find(self, **kwargs):
         url = self._urlexpand(endpoint)
         data = {}
         if "queryMode" in kwargs:
```
Comment (v411e) [inline src/sepsesam/api.py]: Is `queryMode` deprecated? If so, we can also delete these lines

File: src/sepsesam/api.py [hunk 3]
```diff
@@ -1507,21 +1510,53 @@ def external_group_get(self, id):
...
+        :param externalId:          The external name (i.e. the LDAP name) of the group. required (String)
+        :param enabled:             True, if the external group should be enabled, false otherwise. required (bool)
+        :param relation:            The list of relations to sesam user groups belonging to this external group. (String or int)
+        :param type:                The type of the external group. Valid values are "NONE", "AD" and "LDAP".
+        :param mtime:               The time at which the notification object was modified at last.
+        :param usercomment:         A comment by the user about the external group.
+        
         """
-        return self._filter(self.external_group_list(), **kwargs)
+        log.debug("Running function")
```
Comment (v411e) [inline src/sepsesam/api.py]: This log is not very expressive

```suggestion
```

**Review by v411e — APPROVED** [review]

File: src/sepsesam/api.py (hunk 2 above)
Comment (DerYannis1) [reply src/sepsesam/api.py]: It is deprecated and can be deleted.
```suggestion
       
```

File: src/sepsesam/api.py [hunk 4]
```diff
@@ -1836,79 +1853,105 @@ def command_event_get(self, id):
...
         return data
 
-    def drive_group_get(self, id):
+    def drive_group_get(self, id = None, name = None):
         """
         Get a drive group
         """
+        if name:
+            id = self.drive_group_resolveDriveGroupToId(name)
+
```
Comment (v411e) [inline src/sepsesam/api.py:1930]: Awesome!

File: t/tests.py [hunk 5]
```diff
@@ -118,6 +118,125 @@ def test_LocationsCreateAndDelete(self):
...
+        api.command_event_create(**{ "id": 0, "name": "testEvent1", "commandName":"eventCommand1", "clientId": 0})
+        
+        #testing event_get()
+        self.assertEqual(api.command_event_get(0)["name"], "testEvent1")
+
+        #creating second event for find() testing
+        api.command_event_create(**{ "id": 1, "name": "testEvent2", "commandName":"eventCommand1", "clientId": 0})
+        #command_event_find not working yet
+        #self.assertTrue(len(api.command_event_find(**{"commandName": "eventCommand1"}))>1)
```
Comment (v411e) [inline t/tests.py]: Does this work now? :)

File: t/tests.py [hunk 6]
```diff
@@ -118,6 +118,125 @@ def test_LocationsCreateAndDelete(self):
...
+
+        #there should only be found one command event after the update
+        api.command_event_update(1, "testEvent2", 0, **{"object":"eventCommand2"})
+        self.assertEqual(api.command_event_get(1)["object"], "eventCommand2")
+
+        #command_event_find not working yet
+        #len(api.command_event_find(**{"commandName": "eventCommand1"}))==1
```
Comment (v411e) [inline t/tests.py]: Does this work now?

**Review by v411e — APPROVED** [review]

**Pull request merged**